### DISCUSSION TITLE
Word doc (.docx) fails to convert to Markdown

### DISCUSSION THREAD

**Discussion by clfriede** [jgm/pandoc#10384] [General]
Hi all.  I wanted to post this here before posting an issue.  Seeing this problem when I'm trying to convert a word document to markdown (Actual filename replaced with <filename> to provide command I used to generate error):

pandoc --extract-media ./images -o <filename>.md <filename>.docx
couldn't unpack docx container: Did not find end of central directory signature

This seems to happen no matter what .docx file I provide.  Tried without --extract-media as well to see if extracting images may have led to the issue.  This is the version information for my pandoc:

pandoc -v
pandoc.exe 3.5
Features: +server +lua
Scripting engine: Lua 5.4
User data directory: C:\Users\clfriede\AppData\Roaming\pandoc
Copyright (C) 2006-2024 John MacFarlane. Web: https://pandoc.org
This is free software; see the source for copying conditions. There is no
warranty, not even for merchantability or fitness for a particular purpose.

Please let me know if there is something else I can try.  If I don't hear anything or the recommendation is to file an issue, I will do that.  Thanks.

**Comment by jgm** [comment]
Interesting. How did you install this?  Was it from our release binaries?

**Reply by clfriede** [reply]
I tried two different ways.  The first time I installed using the 3.5 MSI.  When I got the error, I saw that there were other methods to install, so I uninstalled using the MSI installer and then reinstalled using the winget method.  Both methods produced the same error.

Regarding the file, this happened to many different files I tried.  I cannot provide those because they contain non-public information, but I could probably create a sample file, make sure it produces the same error and then provide that.  I will post it here once I do.

**Comment by jgm** [comment]
(There was a bug in zip-archive that caused this but it got fixed a while back. Perhaps the pandoc you're using got compiled against an older version of zip-archive -- or maybe it's a different issue.)

**Comment by jgm** [comment]
If you can upload the docx (or email it to me) I could take a look.

**Comment by clfriede** [comment]
@jgm posting two files here.  The first is a sample document that seems to work ok when I run pandoc against it with the arguments I posted above.  It's called Document.docx.  

The second file is one that fails that can be released publicly.  It's called AvereOS_Upgrade_Template.docx.

[Document.docx](https://github.com/user-attachments/files/17739548/Document.docx)
[AvereOS_Upgrade_Template.docx](https://github.com/user-attachments/files/17739553/AvereOS_Upgrade_Template.docx)

What's strange to me is that the second file was, many moons ago, created by using the same template that I used to create the first file today.  

If there's anything else I can provide to help, please let me know.

**Reply by jgm** [reply]
Thank you. I can reproduce the problem.

Note:  not even the `unzip` utility can read this:
```
%  unzip -d Avero ~/Downloads/AvereOS_Upgrade_Template.docx 
Archive:  /Users/jgm/Downloads/AvereOS_Upgrade_Template.docx
  End-of-central-directory signature not found.  Either this file is not
  a zipfile, or it constitutes one disk of a multi-part archive.  In the
  latter case the central directory and zipfile comment will be found on
  the last disk(s) of this archive.
unzip:  cannot find zipfile directory in one of /Users/jgm/Downloads/AvereOS_Upgrade_Template.docx or
        /Users/jgm/Downloads/AvereOS_Upgrade_Template.docx.zip, and cannot find /Users/jgm/Downloads/AvereOS_Upgrade_Template.docx.ZIP, period.
```
Note that docx files are just zip containers, so unzip should work.

Second thing to note: when I try to open this in Word, I get

<img width="421" alt="image" src="https://github.com/user-attachments/assets/a59edf3b-3358-4d2d-90f8-6cc103f048dd">

So, I'm guessing that Microsoft has put some kind of protection on this. It is not a zip file (doesn't have the right kind of header).  There's no way pandoc can deal with the MS authentication, so I think we're at a dead end here.

**Reply by clfriede** [reply]
Ok, I think I may know what may be going on.  What is strange though is that it doesn't work with my permissions as I can see those files without issue and can also edit them.  I'll try a few things here and let you know.

**Reply by clfriede** [reply]
Figured it out.  Recent security change caused the data classification for these documents to change in a way that prevents pandoc from being able to access them.  Not entirely sure why, but when I changed it back to its old classification, it worked without issue.  I think I'm ok from here.  Thanks for all of the help @jgm!
